Mathlib/Data/ListM/Basic.lean

```diff
@@ -248,21 +311,79 @@ unsafe def bind {α β : Type u} : ListM m α → (α → ListM m β) → ListM
 #align tactic.mllist.bind_ ListM.bind
 
 /-- Convert any value in the monad to the singleton monadic lazy list. -/
-unsafe def monad_lift {α} (x : m α) : ListM m α :=
+unsafe def monadLift {α} (x : m α) : ListM m α :=
 cons <| (flip Prod.mk nil ∘ some) <$> x
-#align tactic.mllist.monad_lift ListM.monad_lift
+#align tactic.mllist.monad_lift ListM.monadLift
+
+/-- Lift the monad of a lazy list. -/
+unsafe def liftM [Monad n] [MonadLift m n] (L : ListM m α) : ListM n α :=
+squash do match ← (uncons L : m _) with
+  | none => pure empty
+  | some (a, L') => pure <| cons do pure (a, L'.liftM)
+
+/-- Given a lazy list in a state monad, run it on some initial state, recording the states. -/
+unsafe def run {σ α : Type u} (L : ListM (StateT.{u} σ m) α) (s : σ) : ListM m (α × σ) :=
```

Comment:
```suggestion
unsafe def runState {σ α : Type u} (L : ListM (StateT.{u} σ m) α) (s : σ) : ListM m (α × σ) :=
```
I imagine run-like methods would be useful for other monad transformers as well.